### PR TITLE
Feat create gym route

### PR DESCRIPTION
**Pull request by danssaul**
- [x] add validations to gym entity
- [x] remove unnecessary tests of repo provided by spring
- [x]  implement validation on service and controller with @transactional 
- [x] remove @Autowired annotation following the good practices 
- [x]  add method to repository interface with Spring Data JPA's method query derivation—often called Spring Magic :)

### CODE REVIEW

File: src/main/java/igym/exceptions/DuplicateGymException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package igym.exceptions;
+
+public class DuplicateGymException extends RuntimeException {
+    public DuplicateGymException(String message) {
+        super(message);
+    }
+}
```
Comment (tomerpeceniski) [inline src/main/java/igym/exceptions/DuplicateGymException.java:7]: Hey Daniel, I suggest moving this class to the package services.exceptions

File: src/main/java/igym/exceptions/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -21,6 +22,29 @@ private Map<String, Object> buildResponseBody(HttpStatus status, String message)
         return body;
     }
 
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<Object> handleValidationException(MethodArgumentNotValidException ex) {
+        Map<String, Object> body = new HashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST.value());
+        body.put("error", HttpStatus.BAD_REQUEST.getReasonPhrase());
```
Comment (tomerpeceniski) [inline src/main/java/igym/exceptions/GlobalExceptionHandler.java]: Hey Daniel, instead of rewriting these lines, you should use the method buildResponseBody()

File: src/main/java/igym/exceptions/GlobalExceptionHandler.java
Comment (tomerpeceniski) [inline src/main/java/igym/exceptions/GlobalExceptionHandler.java:1]: Hey Daniel, remember to put this class inside controller.exceptions before merging

File: src/test/java/igym/entities/GymTest.java
Comment (tomerpeceniski) [inline src/test/java/igym/entities/GymTest.java:1]: I think it's missing a test for the case when name = null, right?

File: src/test/java/igym/services/GymServiceTest.java [hunk 3]
```diff
@@ -1,44 +1,75 @@
...
+
+    @Test
+    @DisplayName("should throw DuplicateGymException when attempting to create a gym with a duplicate name")
+    void testAlreadyCreatedGymName() {
+
+        Gym gym = new Gym("CrossFit Gym");
+
+        when(gymRepository.existsByName(gym.getName())).thenReturn(true);
+        when(gymRepository.save(any(Gym.class)))
+                .thenThrow(new DuplicateGymException("A gym with the name 'CrossFit Gym' already exists."));
```
Comment (tomerpeceniski) [inline src/test/java/igym/services/GymServiceTest.java]: This line is not usefull, you can delete it and the test will pass. The rest of the test is good

**Review by tomerpeceniski — COMMENTED** [review]
good job, i made some comments but the core is good

**Review by tomerpeceniski — APPROVED** [review]
very good, sendão! just dont forget to put the exception files in the right places, as we had talked!
